src/app/barter/components/barter-offer/barter-offer.component.ts

```diff
@@ -0,0 +1,83 @@
+import {Component, OnInit, ViewChild} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {User} from '../../../../entities/user.model';
+import {Item} from '../../../../entities/item.model';
+import {getMyItems, ItemsState} from '../../../items/reducers/items.reducer';
+import {takeUntil} from 'rxjs/operators';
+import {ObservableListener} from '../../../components/observable-listener';
+import {Store} from '@ngrx/store';
+import {AuthenticateService} from '../../../user/services/authentication.service';
+import {ItemComponent} from '../../../items/components/item/item.component';
+import {getMyTransactions, TransactionsState} from '../../reducers/transactions.reducer';
+import {createTransaction, createTransactionFail, initiateTransactions} from '../../actions/transactions.actions';
+import {Transaction} from '../../../../entities/transaction.model';
+import firebase from 'firebase';
+
+@Component({
+  selector: 'app-barter-offer',
+  templateUrl: './barter-offer.component.html',
+  styleUrls: ['./barter-offer.component.scss']
+})
+export class BarterOfferComponent extends ObservableListener implements OnInit {
+  user: User;
+  offerItem: Item;
+  myItems: Item[] = [];
+  selectedItem: Item;
+  private myTransactions: Transaction[];
+  @ViewChild('myItem') myItem!: ItemComponent;
+
+
+  constructor(private activatedRoute: ActivatedRoute,
+              private store$: Store<ItemsState>,
+              private transactionsStore$: Store<TransactionsState>,
+              private authService: AuthenticateService) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.transactionsStore$.dispatch(initiateTransactions());
+    this.transactionsStore$.select(getMyTransactions(this.authService.getUserFromLocalStorage().id))
```

Comment:
Wouldn't it be better if you would bring it from the store instead of the local storage?

Reply:
done